docs/devguide/getting_started.rst

```diff
@@ -0,0 +1,1118 @@
+
+If additional samples are available, the service calls this function again.
+However, reading values a single sample at a time is not the most efficient way to process incoming data.
+The Data Reader interface provides a number of different options for processing data in a more efficient manner.
+We discuss some of these operations in Section :ref:`getting_started--data-handling-optimizations`.
+
+.. _getting_started--cleaning-up-in-opendds-clients:
+
+Cleaning up in OpenDDS Clients
+==============================
+
+..
+    Sect<2.1.6>
+
+After we are finished in the publisher and subscriber, we can use the following code to clean up the OpenDDS-related objects:
+
+.. code-block:: cpp
+
+        participant->delete_contained_entities();
+        dpf->delete_participant(participant);
+        TheServiceParticipant->shutdown ();
+
+The domain participant’s ``delete_contained_entities()`` operation deletes all the topics, subscribers, and publishers created with that participant.
+Once this is done, we can use the domain participant factory to delete our domain participant.
+
+Since the publication and subscription of data within DDS is decoupled, data is not guaranteed to be delivered if a publication is disassociated (shutdown) prior to all data that has been sent having been received by the subscriptions.
+If the application requires that all published data be received, the ``wait_for_acknowledgments()`` operation is available to allow the publication to wait until all written data has been received.
+Data readers must have a ``RELIABLE`` setting for the ``RELIABILITY`` QoS (which is the default) in order for ``wait_for_acknowledgments()`` to work.
+This operation is called on individual ``DataWriters`` and includes a timeout value to bound the time to wait.
+The following code illustrates the use of ``wait_for_acknowledgments()`` to block for up to 15 seconds to wait for subscriptions to acknowledge receipt of all written data:
+
+.. code-block:: cpp
+
+      DDS::Duration_t shutdown_delay = {15, 0};
+      DDS::ReturnCode_t result;
+      result = writer->wait_for_acknowledgments(shutdown_delay);
+      if( result != DDS::RETCODE_OK) {
+        std::cerr << "Failed while waiting for acknowledgment of "
+                  << "data being received by subscriptions, some data "
+                  << "may not have been delivered." << std::endl;
+      }
+
+.. _getting_started--running-the-example:
+
+Running the Example
+===================
+
+..
+    Sect<2.1.7>
+
+We are now ready to run our simple example.
+Running each of these commands in its own window should enable you to most easily understand the output.
+
+First we will start a ``DCPSInfoRepo`` service so our publishers and subscribers can find one another.
+
+.. note:: This step is not necessary if you are using peer-to-peer discovery by configuring your environment to use RTPS discovery.
+
+The ``DCPSInfoRepo``  executable is found in :ghfile:`bin`.
```

Comment:
I'm not sure this makes sense as a link to the GitHub repo since it's an artifact of the build